File: src/pojo/VacinaNaoPerecivel.java
package pojo;

/**
 * POJO 6 — VacinaNaoPerecivel
 * Subclasse de ProdutoBiologico.
 * Vacinas liofilizadas ou termotolerantes — armazenamento à temperatura ambiente.
 */
public class VacinaNaoPerecivel extends ProdutoBiologico {

    private String formaFarmaceutica; // Ex: "Liofilizado", "Suspensão termotolerante"
    private int    prazoValidadeMeses;// Validade em meses
    private double temperaturaMax;   // Temperatura máxima de armazenamento (°C)

    public VacinaNaoPerecivel() {}

    public VacinaNaoPerecivel(int id, String nome, double preco, String fabricante,
                              String registroMapa, String especieAlvo, String viaAdministracao,
                              String tipoAgente, String sorotipo, int numDoses,
                              String formaFarmaceutica, int prazoValidadeMeses,
                              double temperaturaMax) {
        super(id, nome, preco, fabricante, registroMapa, especieAlvo, viaAdministracao,
              tipoAgente, sorotipo, numDoses);
        this.formaFarmaceutica  = formaFarmaceutica;
        this.prazoValidadeMeses = prazoValidadeMeses;
        this.temperaturaMax     = temperaturaMax;
    }

    // ── Getters / Setters ────────────────────────────────────────────────────

    public String getFormaFarmaceutica()                             { return formaFarmaceutica; }
    public void   setFormaFarmaceutica(String formaFarmaceutica)     { this.formaFarmaceutica = formaFarmaceutica; }

    public int    getPrazoValidadeMeses()                            { return prazoValidadeMeses; }
    public void   setPrazoValidadeMeses(int p)                       { this.prazoValidadeMeses = p; }

    public double getTemperaturaMax()                                { return temperaturaMax; }
    public void   setTemperaturaMax(double t)                        { this.temperaturaMax = t; }

    @Override
    public String toString() {
        return "VacinaNaoPerecivel{" +
               "id=" + getId() +
               ", nome='" + getNome() + '\'' +
               ", forma='" + formaFarmaceutica + '\'' +
               ", validadeMeses=" + prazoValidadeMeses +
               ", tempMax=" + temperaturaMax + "°C}";
    }
}

File: vet/cliente/src/VacinaNaoPerecivel.cpp
#include "VacinaNaoPerecivel.hpp"

VacinaNaoPerecivel::VacinaNaoPerecivel(){}

VacinaNaoPerecivel::VacinaNaoPerecivel(int id, const string& nome, double preco, const string& fabricante, 
    const string& regM, const string& espA, const string& vAdm, 
    const string& tipoAgente, const string& sorotipo, int numDoses, 
    const string& formaFarmaceutica, int prazoValidadeMeses, double temperaturaMax)
    : ProdutoBiologico(id, nome, preco, fabricante, regM, espA, vAdm, tipoAgente, sorotipo, numDoses) ,
    formaFarmaceutica(formaFarmaceutica), prazoValidadeMeses(prazoValidadeMeses), temperaturaMax(temperaturaMax) {};

const string& VacinaNaoPerecivel::getFormaFarmaceutica() const{
    return this->formaFarmaceutica;
}

void VacinaNaoPerecivel::setFormaFarmaceutica(const string& f) {
    this->formaFarmaceutica = f;
}

int VacinaNaoPerecivel::getprazoValidadeMeses() const{
    return this->prazoValidadeMeses;
}

void VacinaNaoPerecivel::setprazoValidadeMeses(int p) {
    this->prazoValidadeMeses = p;
}

double VacinaNaoPerecivel::getTemperaturaMax() const{
    return this->temperaturaMax;
}

void VacinaNaoPerecivel::setTemperaturaMax(double tempMax) {
    this->temperaturaMax = tempMax;
}

string VacinaNaoPerecivel::toString() const{
    return "VacinaNaoPerecivel{id=" + to_string(this->getId()) + 
    ", nome='" + this->getNome() + 
    "', forma='" + this->formaFarmaceutica + 
    "', validadeMeses=" + to_string(this->prazoValidadeMeses) + 
    ", tempMax=" + to_string(this->temperaturaMax) + 
    "°C}";
}